### ISSUE TITLE
'disk_size_gb' property is blank for virtual machines

### ISSUE THREAD

**Issue by krimkus** [Azure/azure-sdk-for-python#672]
After fetching virtual machine info using `ComputeClient.virtual_machines.get()`, I always get a null value for `storage_profile.os_disk.disk_size_gb`. Is there an alternate way of getting that property?

**Comment by lmazuel** [comment]
Yes, it seems to be a Rest Api issue. [I had seen this too when I made the sample](https://github.com/Azure-Samples/virtual-machines-python-manage/blob/master/example.py#L174).

I didn't try yet, but I guess it should be possible to do something using the storage mgmt lib.

**Comment by lmazuel** [comment]
Did you figure out a way to get the size?

**Comment by lmazuel** [comment]
Hi @krimkus ,

I got a response from the VM team:
> CRP doesn’t return the value for diskSizeGB unless it was input by the customer in the firstplace.

So, it's not a bug, it gives you the initial value you wrote. It might be enough for your situation if you are able to fill it at VM creation. Otherwise, you will have to do it another way. Did you try with the storage mgmt lib?

**Comment by krimkus** [comment]
By 'storage mgmt lib', do you mean the [Azure Storage SDK](https://github.com/Azure/azure-storage-python) or the [Storage Resource Management section ](https://github.com/Azure/azure-sdk-for-python/tree/master/azure-mgmt-storage) of the full Python SDK? I'm assuming you mean the latter, but I have not found that info via either. Do you know what model would provide that info? I don't see any fields related to disk size on the Resource model, and I don't see how to get Usage for a particular disk or resource. Am I looking in the wrong places?

**Comment by lmazuel** [comment]
Hi @krimkus 

I make a few tests and find how to handle this (valid for 2.0.0rc5)
```python
# First, test the attribute, won't be None if you resized at least once
In [13]: vm.storage_profile.os_disk.disk_size_gb
Out[13]: 89

# If this attribute is None or empty string, disk size depends of the image used to build the VM
# Then, we have to use the storage account

# Step1: Get the VHD address, to parse storage_account and blob name
In [16]: vm.storage_profile.os_disk.vhd.uri
Out[16]: 'https://testvmsize123456789.blob.core.windows.net/vhds/TestVmSize2016791625.vhd'

storage_account = 'testvmsize123456789'
blob_name = 'TestVmSize2016791625.vhd'

# Step2: Get a valid storage data Python client
from azure.mgmt.storage import StorageManagementClient
from azure.storage import CloudStorageAccount

storage_mgmt_client = StorageManagementClient(credentials, subscription_id)
storage_keys = storage_mgmt_client.storage_account.list_keys('TestVmSize', storage_account)
storage_key = storage_keys.keys[0].value
storage_data_client = CloudStorageAccount(storage_account, storage_key)
storage_blob_client = storage_data_client.create_block_blob_service()

# Step3: Read the value
blob_properties = storage_blob_client.get_blob_properties('vhds', blob_name)

In [28]: blob_properties.properties.content_length
Out[28]: 95563022848
```

I hope this helps, please tell me what you think about it.

Thanks!

**Comment by krimkus** [comment]
That works pretty well, thanks. I had to modify it in a few places (in case anyone else uses this example):

The `storage_mgmt_client.storage_account` property should be `storage_mgmt_client.storage_accounts`

Since `storage_mgmt_client.storage_accounts.list_keys()` requires the resource group of the storage account, I had to fetch that. It doesn't appear to be a property of the vm, and not part of the vhd.uri, I had to loop over all possible storage accounts in the subscription and find the one being used to determine its resource group:

```
storage_resource_group = None
for sa in storage_mgmt_client.storage_accounts.list():
    sa_id_parts = sa.id.split('/')
    if sa_id_parts[-1] == storage_account:
        storage_resource_group = sa_id_parts[4]
```
Then, I can pass that to .list_keys() instead of `'TestVmSize'` used in the example.

Thanks again for the help.
